sklearn/feature_extraction/_dict_vectorizer.py

```diff
@@ -98,6 +119,27 @@ def __init__(self, *, dtype=np.float64, separator="=", sparse=True,
         self.sparse = sparse
         self.sort = sort
 
+    def _add_iterable_element(self, f, v, feature_names, vocab, fitting=True,
+                              transforming=False, indices=None, values=None):
+        """Add feature names for iterable of strings"""
+        for vv in v:
+            if isinstance(vv, str):
+                feature_name = "%s%s%s" % (f, self.separator, vv)
+                vv = 1
+            else:
+                raise TypeError(f'Unsupported type {type(vv)} in iterable '
+                                'value. Only iterables of string are '
+                                'supported.')
+            if fitting and feature_name not in vocab:
+                vocab[feature_name] = len(feature_names)
+                feature_names.append(feature_name)
+
+            if transforming and feature_name in vocab:
+                indices.append(vocab[feature_name])
+                values.append(self.dtype(vv))
```

Comment:
I've just realised that if a string is repeated in the iterable, then this will increment it twice (since Scipy sparse matrices sum values over repeated indices).

I think this behaviour is okay, but we need to document it differently. Or we need to forbid this behaviour by raising an error with duplicate strings.

Reply:
I'm slightly in favour of forcing unique strings...

Reply:
> I'm slightly in favour of forcing unique strings...

It's certainly the one that is easiest to change later, since we would be turning an error into a supported case.

Reply:
> It's certainly the one that is easiest to change later, since we would be turning an error into a supported case.

I was also thinking about the use case described in #6045: attaching categories to movies, or labels to pull requests, we probably don't want to duplicate them... but, sure, who knows, at least it will be easy to get rid of the error...

Reply:
I think it's fine to count repeated labels twice. This is consistent with what `CountVectorizer` for strings with repeated tokens.

I a way, `DictVectorizer` would behave as a natural complement for `CountVectorizer`  when the tokenization step would have already been performed as a preprocessing step.

Reply:
Also it's easy for the users to filter duplicates in their inputs by using Python `set` instead of lists if they don't want this.